src/vivarium_conic_sqlns/components/iron_deficiency.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import pandas as pd
+
+from vivarium_public_health.utilities import EntityString
+from vivarium_public_health.risks.data_transformations import get_distribution
+
+
+class Hemoglobin:
+
+    configuration_defaults = {
+        "iron_deficiency": {
+            "exposure": 'data',
+            "rebinned_exposed": [],
+            "category_thresholds": [],
+        }
+    }
+
+    def __init__(self):
+        self.name = "hemoglobin_level"
+        self.risk = EntityString("risk_factor.iron_deficiency")
+
+    def setup(self, builder):
+        self.hemoglobin_distribution = get_distribution(builder, self.risk)
+        builder.components.add_components([self.hemoglobin_distribution])
+
+        self.pop_view = builder.population.get_view(['alive'])
+
+        self.anemia_thresholds = builder.lookup.build_table(get_anemia_thresholds(), key_columns=[],
+                                                            parameter_columns=[('age',
+                                                                                'age_group_start',
+                                                                                'age_group_end')],
+                                                            value_columns=['severe_threshold', 'moderate_threshold',
+                                                                           'mild_threshold'])
+
+        self.randomness = builder.randomness.get_stream('initial_hemoglobin_propensity')
+        self._hemoglobin = pd.Series()
+
+        self.hemoglobin = builder.value.register_value_producer('hemoglobin',
+                                                                source=lambda index: self._hemoglobin[index])
+
+        self._disability_weight = builder.lookup.build_table(get_iron_deficiency_disability_weight(builder))
+        self.disability_weight = builder.value.register_value_producer('iron_deficiency.disability_weight',
+                                                                      source=self.compute_disability_weight)
+        builder.value.register_value_modifier('disability_weight', modifier=self.disability_weight)
+
+        builder.population.initializes_simulants(self.on_initialize_simulants)
+
+    def on_initialize_simulants(self, pop_data):
+        propensity = self.randomness.get_draw(pop_data.index)
+        new_sims_hemoglobin = pd.Series(self.hemoglobin_distribution.ppf(propensity), index=pop_data.index)
+        self._hemoglobin = self._hemoglobin.append(new_sims_hemoglobin)
+        self.anemia_thresholds(pop_data.index)
```

Comment:
this lookup table call is hanging. I think since anemia is determined on the fly in compute_disability_weight it should be removed